src/test/kotlin/no/nav/syfo/api/endpoints/OppfolgingsplanEndpointsTest.kt

```diff
@@ -0,0 +1,153 @@
+package no.nav.syfo.api.endpoints
+
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.jackson.*
+import io.ktor.server.testing.*
+import io.mockk.clearMocks
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.slot
+import no.nav.syfo.ExternalMockEnvironment
+import no.nav.syfo.UserConstants.ARBEIDSTAKER_PERSONIDENT
+import no.nav.syfo.UserConstants.ARBEIDSTAKER_PERSONIDENT_VEILEDER_NO_ACCESS
+import no.nav.syfo.UserConstants.NARMESTELEDER_FNR
+import no.nav.syfo.UserConstants.VEILEDER_IDENT
+import no.nav.syfo.UserConstants.VIRKSOMHETSNUMMER
+import no.nav.syfo.api.generateJWT
+import no.nav.syfo.api.model.*
+import no.nav.syfo.api.testApiModule
+import no.nav.syfo.domain.Foresporsel
+import no.nav.syfo.domain.Personident
+import no.nav.syfo.infrastructure.NAV_PERSONIDENT_HEADER
+import no.nav.syfo.infrastructure.database.dropData
+import no.nav.syfo.util.configure
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+object OppfolgingsplanEndpointsTest {
+    val urlOppfolgingsplan = "/api/internad/v1/oppfolgingsplan"
+    val externalMockEnvironment = ExternalMockEnvironment.instance
+    val database = externalMockEnvironment.database
+    val varselProducer = externalMockEnvironment.varselProducer
+
+    fun ApplicationTestBuilder.setupApiAndClient(): HttpClient {
+        application {
+            testApiModule(
+                externalMockEnvironment = ExternalMockEnvironment.instance,
+            )
+        }
+        val client =
+            createClient {
+                install(ContentNegotiation) {
+                    jackson { configure() }
+                }
+            }
+        return client
+    }
+
+    val validToken =
+        generateJWT(
+            audience = externalMockEnvironment.environment.azure.appClientId,
+            issuer = externalMockEnvironment.wellKnownInternalAzureAD.issuer,
+            navIdent = VEILEDER_IDENT.value,
+        )
+
+    val personIdent = ARBEIDSTAKER_PERSONIDENT.value
+    val personIdentNoAccess = ARBEIDSTAKER_PERSONIDENT_VEILEDER_NO_ACCESS.value
+
+    @BeforeEach
+    fun beforeEachTest() {
+        database.dropData()
+        clearMocks(varselProducer)
+        val foresporselSlot = slot<Foresporsel>()
+        every { varselProducer.sendNarmesteLederVarsel(capture(foresporselSlot)) }.answers {
+            Result.success(foresporselSlot.captured)
+        }
+    }
+
+    @Test
+    fun `Successfully creates a new foresporsel with varsel`() {
+        testApplication {
+            val client = setupApiAndClient()
```

Comment:
Er dette noe som må gjøres for hver test? 🤔 Tenkte vi kunne unngå litt boilerplate om vi kunne instantiere dette et felles sted i filen 😊

Reply:
Vet ikke om noen bedre måte å gjøre det på dessverre. Implementasjonen av `setupApiAndClient()` kan man flytte ut av testklassen, men det hjelper ikke noe i dette tilfellet.